### PR TITLE
Option to keep or drop some dice from those rolled

### PR DESCRIPTION
**Pull request by olavmueller**

### CODE REVIEW

**Comment by justinian** [comment]
Oh, thanks for this PR! Great idea for this addition. Looks pretty much good to go, just a couple minor comments:

File: std.go [hunk 1]
```diff
@@ -5,25 +5,29 @@ import (
...
 	Total int
 }
 
 func (r StdResult) String() string {
-	return fmt.Sprintf("%d %v", r.Total, r.Rolls)
+	return fmt.Sprintf("%d %v (%v)", r.Total, r.Rolls, r.Dropped)
 }
 
 func (StdRoller) Roll(matches []string) (RollResult, error) {
+	fmt.Println( matches )
```
Comment (justinian) [inline std.go]: This looks like it might have been for testing? Let's pull this out.

File: std.go [hunk 2]
```diff
@@ -34,25 +38,61 @@ func (StdRoller) Roll(matches []string) (RollResult, error) {
...
+			return nil, err
+		}
+		num = int(number)
+		keep = matches[4]
+	}
+	
 	result := StdResult{
 		basicRollResult: basicRollResult{matches[0]},
 		Rolls:           make([]int, dice),
+		Dropped:		 nil,
```
Comment (justinian) [inline std.go]: Looks like a tabs/spaces mismatch here - can you run the file through `gofmt`?

File: std.go [hunk 3]
```diff
@@ -34,25 +38,61 @@ func (StdRoller) Roll(matches []string) (RollResult, error) {
...
 		result.Rolls[i] = roll
 	}
+
+	sort.Ints( result.Rolls )
+	size := len( result.Rolls )
+	
+	if keep == "kh" {
+		result.Dropped = result.Rolls[:size-num]
+		result.Rolls = result.Rolls[size-num:]
+	}
```
Comment (justinian) [inline std.go]: Not a huge deal, but you could make this a `switch` statement, which might look cleaner. I'm fine with it either way though, to be honest.

**Review by justinian — CHANGES_REQUESTED** [review]

File: std.go (hunk 1 above)
Comment (olavmueller) [reply std.go]: Sure. Sorry.

**Pull request closed without merging**